pkg/cmd/provisioning/azure/create_oidc_issuer.go

```diff
@@ -0,0 +1,588 @@
+		return &getResourceGroupResp.ResourceGroup, nil
+	}
+
+	// Create resource group
+	createResourceGroupResp, err := client.ResourceGroupsClient.CreateOrUpdate(
+		context.Background(),
+		resourceGroupName,
+		armresources.ResourceGroup{
+			Location: to.Ptr(region),
+			Tags:     mergedResourceTags,
+		},
+		nil)
+	if err != nil {
+		return nil, err
+	}
+	verb := "Updated"
+	if needToCreateResourceGroup {
+		verb = "Created"
+	}
+	log.Printf("%s resource group %s", verb, *createResourceGroupResp.ResourceGroup.ID)
+	return &createResourceGroupResp.ResourceGroup, nil
+}
+
+// ensureStorageAccount ensures that a storage account with storageAccountName exists within the provided resource group, region and subscription
+// resourceTags will be updated to match those provided to ensureStorageAccount if found to be different on an existing storage account.
+func ensureStorageAccount(client *azureclients.AzureClientWrapper, storageAccountName, resourceGroupName, region string, resourceTags map[string]string) (*armstorage.Account, error) {
+	listAccounts := client.StorageAccountClient.NewListByResourceGroupPager(resourceGroupName, &armstorage.AccountsClientListByResourceGroupOptions{})
+	list := make([]*armstorage.Account, 0)
+	for listAccounts.More() {
+		pageResponse, err := listAccounts.NextPage(context.Background())
+		if err != nil {
+			return nil, err
+		}
+		list = append(list, pageResponse.AccountListResult.Value...)
+	}
+	needToCreateStorageAccount := true
+	needToUpdateStorageAccount := false
+	mergedResourceTags := map[string]*string{}
+	var existingStorageAccount *armstorage.Account
+	for _, storageAccount := range list {
+		if *storageAccount.Name == storageAccountName {
+			existingStorageAccount = storageAccount
+			// Validate that existing storage account is in requested region
+			if *storageAccount.Location != region {
+				return nil, fmt.Errorf("found existing storage account %s in unexpected region=%s, requested region=%s",
+					*storageAccount.ID,
+					*storageAccount.Location,
+					region)
+			}
+			needToCreateStorageAccount = false
+			if storageAccount.Tags != nil {
+				for key, value := range storageAccount.Tags {
+					mergedResourceTags[key] = value
+				}
+			}
+			for tagKey, tagValue := range resourceTags {
+				val, ok := mergedResourceTags[tagKey]
+				// Current tags don't contain expected tag keys and values
+				if !ok || (ok && *val != tagValue) {
+					mergedResourceTags[tagKey] = &tagValue
+					needToUpdateStorageAccount = true
+				}
+			}
+		}
+	}
+
+	if !needToCreateStorageAccount && !needToUpdateStorageAccount {
+		// Found and validated existing storage account, return
+		log.Printf("Found existing storage account %s", *existingStorageAccount.ID)
+		return existingStorageAccount, nil
+	}
+
+	// Create storage account
+	if needToCreateStorageAccount {
+		createStorageAccountTags := map[string]*string{}
+		for key, value := range resourceTags {
```

Comment:
Code review: If we merge the tags above outside of the `if` in which we locate the storage account by name, then we don't have to rebuild the tag list here... we could already have made it.